third_party/repositories/scala_3_1.bzl

```diff
@@ -99,6 +99,10 @@ artifacts = {
             "@io_bazel_rules_scala_scala_library",
         ],
     },
+    "org_scala_sbt_compiler_interface": {
```

Comment:
What should be better in your opinion - add this artifact for versions that don't need that (3.1.0 =< version < 3.4.0) to stay concise with [artifact_ids](https://github.com/bazelbuild/rules_scala/pull/1603/files#diff-219f92bb35f88485a21a0615e73a715488fa38dbeeda0eb7b9abd251c18573d9L126) or modify `def _artifact_ids(scala_version):`? If modifying, I can do that in next PR.

Reply:
I'd say keep it as is and do not complicate  `def _artifact_ids(scala_version)` any further. I guess those who use scala 3 will quickly move to never version and older ones will become obsolete.